cpp/src/arrow/acero/aggregate_benchmark.cc

```diff
@@ -866,5 +887,61 @@ BENCHMARK(TDigestKernelDoubleMedian)->Apply(QuantileKernelArgs);
 BENCHMARK(TDigestKernelDoubleDeciles)->Apply(QuantileKernelArgs);
 BENCHMARK(TDigestKernelDoubleCentiles)->Apply(QuantileKernelArgs);
 
+//
+// Segmented Aggregate
+//
+
+static void BenchmarkSegmentedAggregate(
+    benchmark::State& state, int64_t num_rows, std::vector<Aggregate> aggregates,
+    const std::vector<std::shared_ptr<Array>>& arguments,
+    const std::vector<std::shared_ptr<Array>>& keys, int64_t num_segment_keys,
+    int64_t num_segments) {
+  ASSERT_GT(num_segments, 0);
+
+  auto rng = random::RandomArrayGenerator(42);
+  auto segment_key = rng.Int64(num_rows, /*min=*/0, /*max=*/num_segments - 1);
+  int64_t* values = segment_key->data()->GetMutableValues<int64_t>(1);
+  std::sort(values, values + num_rows);
+  // num_segment_keys copies of the segment key.
+  ArrayVector segment_keys(num_segment_keys, segment_key);
```

Comment:
This is not much better than before, is it? I would expect something like:
```suggestion
  ArrayVector segment_keys(num_segment_keys);
  for (auto& segment_key : segment_keys) {
    segment_key = rng.Int64(num_rows, /*min=*/0, /*max=*/num_segments - 1);
    int64_t* values = segment_key->data()->GetMutableValues<int64_t>(1);
    std::sort(values, values + num_rows);
  }
```

Reply:
Ah, I see your point. It is only that I want to make the number of segments to be exactly as specified. Combining independently-random keys, even with the same distribution, will make the number of segments (potentially, much) bigger.

Reply:
But that's also much more realistic, isn't it?

Reply:
That said, as you prefer.

Reply:
> But that's also much more realistic, isn't it?

I think having exact number of segments would help on diagnostics of performance issues than introducing yet another level of randomness, so I'd rather keep it as is :)